### PR TITLE
Don't include getauxval in sanitizer builds

### PR DESCRIPTION
**Pull request by Algunenano**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Remove custom getauxval on sanitizer builds. This increase the GLIBC dependency on x86, of those official builds only, to 2.16 (was 2.4)

It seems it's not compatible with sanitizer re-entry to handle high-bit ASLR and it causes runtime problems even with updated LLVM libs:

```
ckHouse/build_msan/src/Server/grpc_protos/clickhouse_grpc.pb.cc /mnt/ch/ClickHouse/build_msan/src/Server/grpc_protos/clickhouse_grpc.grpc.pb.h /mnt/ch/ClickHouse/build_msan/src/Server/grpc_protos/clickhouse_grpc.grpc.pb.cc 
cd /mnt/ch/ClickHouse/build_msan/src/Server/grpc_protos && /mnt/ch/ClickHouse/build_msan/contrib/google-protobuf-cmake/protoc --cpp_out /mnt/ch/ClickHouse/build_msan/src/Server/grpc_protos --grpc_out /mnt/ch/ClickHouse/build_msan/src/Server/grpc_protos --plugin=protoc-gen-grpc=/mnt/ch/ClickHouse/build_msan/contrib/grpc-cmake/grpc_cpp_plugin -I /mnt/ch/ClickHouse/src/Server/grpc_protos /mnt/ch/ClickHouse/src/Server/grpc_protos/clickhouse_grpc.proto
MemorySanitizer:DEADLYSIGNAL
==58641==ERROR: MemorySanitizer: SEGV on unknown address 0x2ffeb6d9bbb0 (pc 0x60b9e94cd073 bp 0x000000000003 sp 0x7ffeb6d9bb90 T58641)
==58641==The signal is caused by a WRITE memory access.
    #0 0x60b9e94cd073 in __auxv_init_procfs /mnt/ch/ClickHouse/base/glibc-compatibility/musl/getauxval.c:93:5
    #1 0x60b9e94ccfe9 in getauxval /mnt/ch/ClickHouse/base/glibc-compatibility/musl/getauxval.c:204:12
    #2 0x60b9e8588fb4 in __sanitizer::ReExec() crtstuff.c
    #3 0x60b9e8604bfc in __msan::InitShadowWithReExec(bool) crtstuff.c
    #4 0x60b9e85a2356 in __msan_init (/mnt/ch/ClickHouse/build_msan/contrib/google-protobuf-cmake/protoc+0x256356) (BuildId: 752f7fa248bce2883327c13f80f808b6ac83e357)
    #5 0x60b9e860b878 in msan.module_ctor main.cc
    #6 0x60b9e94ce56c in __libc_csu_init (/mnt/ch/ClickHouse/build_msan/contrib/google-protobuf-cmake/protoc+0x118256c) (BuildId: 752f7fa248bce2883327c13f80f808b6ac83e357)
    #7 0x7a7e45b69ea3 in __libc_start_main /usr/src/debug/glibc/glibc/csu/../csu/libc-start.c:343:6
    #8 0x60b9e85787cd in _start (/mnt/ch/ClickHouse/build_msan/contrib/google-protobuf-cmake/protoc+0x22c7cd) (BuildId: 752f7fa248bce2883327c13f80f808b6ac83e357)

MemorySanitizer can not provide additional info.
SUMMARY: MemorySanitizer: SEGV /mnt/ch/ClickHouse/base/glibc-compatibility/musl/getauxval.c:93:5 in __auxv_init_procfs
==58641==ABORTING
ninja: build stopped: subcommand failed.
```

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-clickhouse-ci-2** [comment]
*This is an automated comment for commit ebe781004c25816e475667320367bebc8c5b2fa6 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/67073/ebe781004c25816e475667320367bebc8c5b2fa6/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/67037/81c19b02e78ea7f2980531732bbaf304622e81fb/docs_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

**Comment by Algunenano** [comment]
This should fail the glibc validation.

File: base/glibc-compatibility/CMakeLists.txt [hunk 1]
```diff
@@ -8,6 +8,41 @@ if (GLIBC_COMPATIBILITY)
...
+        # The source of the issue above is that, at this point in time during __msan_init, we can't really do much as
+        # most global variables aren't initialized or available yet, so we can't initiate the auxiliary vector.
+        # Normal glibc / musl getauxval doesn't have this problem since they initiate their auxval vector at the very
+        # start of __libc_start_main (just keeping track of argv+argc+1), but we don't have such option (otherwise
+        # this complexity of reading "/proc/self/auxv" or using __environ would not be necessary).
+        #
+        # To avoid this crashes on the re-exec call (see above how it would fail when creating `aux`, and if we used
+        # __auxv_init_environ then it would SIGSEV on READing `__environ`) we don't include getauxval for the sanitizers
+        # This means that those builds will depend on GLIBC 2.16 instead of 2.4 in x86_64
+        list(REMOVE_ITEM glibc_compatibility musl/getauxval.c)
```
Comment (Algunenano) [inline base/glibc-compatibility/CMakeLists.txt:43]: Did not work for some reason

**Comment by Algunenano** [comment]
Closing since we want to keep compatibility with old systems to be able to download from CI and run tests in old machines

**Pull request closed without merging**